src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -659,13 +702,63 @@ void Connection::handle_reconnect_wait(Status status)
         initiate_reconnect(); // Throws
 }
 
+struct Connection::WebSocketObserverImpl : public sync::WebSocketObserver {
```

Comment:
Nice! Technically, Connection doesn't even need to implement the `WebSocketObserver` interface anymore...

I was thinking a name like `ConnectionWrapper` would better describe its purpose, but no need to change.

Reply:
I've change Connection to not inherit from WebSocketObserver, although all the functions remain the same, and I've renamed this to WebSocketObserverShim to more closely describe its function since it's a shim between Connection and the Websocket code.